src/index.ts

```diff
@@ -1590,7 +1589,23 @@ export class BigQuery extends common.Service {
       // The Job is important for the `queryAsStream_` method, so a new query
       // isn't created each time results are polled for.
       options = extend({job}, options);
-      job!.getQueryResults(options, callback as QueryRowsCallback);
+
+      // table#getRows uses listTableData endpoint, which is a faster method
+      // to read rows of results.
+
+      job!.getQueryResults(options, (err, rows) => {
+        if (!err) {
+          if (rows!.length !== 0) {
```

Comment:
This throws me off a little bit, does this imply that the best way to go is to poll the `getQueryResults` API until we get a single page of results and only then should we call `listTableData`?

Reply:
Yeah, that's what Seth told me they do in the other languages and how I should handle it to be consistent.